employee-module/src/main/java/employee/exception/handler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,137 @@
+package employee.exception.handler;
+
+import employee.exception.DepartmentNameAlreadyExistsException;
+import employee.exception.DepartmentNotFoundException;
+import employee.exception.EmailAlreadyExistsException;
+import employee.exception.UserNotFoundException;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Global exception handler for employee module.
+ * Centralizes exception handling and provides consistent error responses.
+ */
+@RestControllerAdvice
+@Slf4j
+public class GlobalExceptionHandler {
+
+    /**
+     * Handles UserNotFoundException.
+     * Returns 404 NOT FOUND status.
+     */
+    @ExceptionHandler(UserNotFoundException.class)
+    public ResponseEntity<ErrorResponse> handleUserNotFoundException(
+            UserNotFoundException ex,
+            HttpServletRequest request) {
+        log.error("User not found: {}", ex.getMessage());
```

Comment:
cały wyjątek przekaż, bo jak będziesz miał NPE to w `ex.getMessage()` będzie `null`